### PR TITLE
feat: add padd target regions rule

### LINKED ISSUE #2: migrate rule GetPaddedBed
rule that will be converted: 
```
rule GetPaddedBed:
    params:
        padding= 100,
        target_bed = load_local(config["table_data"]["target_regions"]),
        script_location = config["run_location"]
    output:
        interval = "work/{seqID}/Results/bam/padded_bait_interval.bed"
    singularity:
        config["singularities"]["get_target"]
    shell:
        """
        python3 {params.script_location}/src/Summary/reform_genomic_region.py \
            --target_bed={params.target_bed} \
            --output_file={output.interval} \
            --padding={params.padding} \
            --format='bed'
        """
```

## scripts:
- [x] move  [reform_genomic_region.py ](https://github.com/JoelAAs/pgx_module/blob/master/src/Summary/reform_genomic_region.py)
- [x]  write unit tests for reform_genomic_region.py
- [ ]  update requirements with dependencies 

script should be located under workflow/scripts and probably utilize  the [snakemake  object](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#external-scripts) to fetch parameters, instead of argparse 

## snakemake rules
- [x]  create snakemake rule that uses the script

### config
extra entries:
- [x] target_bed
- [x] padding

## Other
Please follow [hydra-genetics guidelines](https://docs.google.com/document/d/1l2v1ItZBTDaI72vQPZcaQzxwVUao78XzIJvGASAAD9E/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by ljmesi**
Add `padd_target_regions` as hydra rule. 
Closes #2 .

### CODE REVIEW

File: workflow/rules/padd_target_regions.smk [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    log:
+        "pgx/padd_target_regions/{sample}_{type}.output.log",
+    benchmark:
+        repeat(
+            "pgx/padd_target_regions/{sample}_{type}.output.benchmark.tsv",
+            config.get("padd_target_regions", {}).get("benchmark_repeats", 1),
+        )
+    threads:
+        config.get("padd_target_regions", {}).get(
+            "threads", config["default_resources"]["threads"])
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:26]:
```suggestion
    threads: config.get("padd_target_regions", {}).get("threads", config["default_resources"]["threads"])
```

File: workflow/rules/padd_target_regions.smk [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    resources:
+        threads=config.get("padd_target_regions", {}).get(
+            "threads", config["default_resources"]["threads"]
+        ),
+        time=config.get("padd_target_regions", {}).get(
+            "time", config["default_resources"]["time"]
+        ),
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:33]:
```suggestion
        time=config.get("padd_target_regions", {}).get("time", config["default_resources"]["time"]),
```

File: workflow/rules/padd_target_regions.smk [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        mem_mb=config.get("padd_target_regions", {}).get(
+            "mem_mb", config["default_resources"]["mem_mb"]
+        ),
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:36]:
```suggestion
        mem_mb=config.get("padd_target_regions", {}).get("mem_mb", config["default_resources"]["mem_mb"]),
```

File: workflow/rules/padd_target_regions.smk [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        mem_per_cpu=config.get("padd_target_regions", {}).get(
+            "mem_per_cpu", config["default_resources"]["mem_per_cpu"]
+        ),
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:39]:
```suggestion
        mem_per_cpu=config.get("padd_target_regions", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
```

File: workflow/rules/padd_target_regions.smk [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        partition=config.get("padd_target_regions", {}).get(
+            "partition", config["default_resources"]["partition"]
+        ),
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:42]:
```suggestion
        partition=config.get("padd_target_regions", {}).get("partition", config["default_resources"]["partition"]),
```

File: workflow/rules/padd_target_regions.smk [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    container:
+        config.get("padd_target_regions", {}).get(
+            "container", config["default_container"]
+        )
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:46]:
```suggestion
        config.get("padd_target_regions", {}).get("container", config["default_container"])
```

File: workflow/rules/padd_target_regions.smk [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    conda:
+        "../envs/padd_target_regions.yaml"
+    message:
+        "{rule}: Padd bed file on pgx/{rule}/{wildcards.sample}_{wildcards.type}_padded_bait_interval.bed"
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:50]:
```suggestion
        "{rule}: padd bed file on pgx/{rule}/{wildcards.sample}_{wildcards.type}_padded_bait_interval.bed"
```
small letter

**Comment by Smeds** [comment]
please also add this to common.smk

```
from hydra_genetics import min_version as hydra_min_version

hydra_min_version("0.11.0")
```

**Comment by Smeds** [comment]
@ljmesi  Great work!

File: workflow/rules/padd_target_regions.smk [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+__copyright__ = "Copyright 2022, Lauri Mesilaakso"
+__email__ = "lauri.mesilaakso@regionostergotland.se"
+__license__ = "GPL-3"
+
+
+rule padd_target_regions:
+    input:
+        target_regions=config.get("padd_target_regions", {}).get("target_regions", ""),
+    output:
+        padded_target_regions="pgx/padd_target_regions/{sample}_{type}_padded_bait_interval.bed",
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:14]:
```suggestion
        padded_target_regions=temp("pgx/padd_target_regions/{sample}_{type}_padded_bait_interval.bed"),
```

File: workflow/rules/padd_target_regions.smk [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+            "pgx/padd_target_regions/{sample}_{type}.output.benchmark.tsv",
+            config.get("padd_target_regions", {}).get("benchmark_repeats", 1),
+        )
+    threads: config.get("padd_target_regions", {}).get("threads", config["default_resources"]["threads"])
+    resources:
+        threads=config.get("padd_target_regions", {}).get("threads", config["default_resources"]["threads"]),
+        time=config.get("padd_target_regions", {}).get("time", config["default_resources"]["time"]),
+        mem_mb=config.get("padd_target_regions", {}).get("mem_mb", config["default_resources"]["mem_mb"]),
+        mem_per_cpu=config.get("padd_target_regions", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
+        partition=config.get("padd_target_regions", {}).get("partition", config["default_resources"]["partition"]),
```
Comment (Smeds) [inline workflow/rules/padd_target_regions.smk:30]:
```suggestion
        mem_mb=config.get("padd_target_regions", {}).get("mem_mb", config["default_resources"]["mem_mb"]),
        mem_per_cpu=config.get("padd_target_regions", {}).get("mem_per_cpu", config["default_resources"]["mem_per_cpu"]),
        partition=config.get("padd_target_regions", {}).get("partition", config["default_resources"]["partition"]),
        threads=config.get("padd_target_regions", {}).get("threads", config["default_resources"]["threads"]),
        time=config.get("padd_target_regions", {}).get("time", config["default_resources"]["time"]),
```

File: workflow/schemas/config.schema.yaml [hunk 10]
```diff
@@ -17,3 +17,20 @@ required:
...
   - resources
   - default_container
+
+padd_target_regions:
+    type: object
+    description: parameters for padd_target_regions
+    properties:
+      benchmark_repeats:
+        type: integer
+        description: set number of times benchmark should be repeated for padd_target_regions
```
Comment (Smeds) [inline workflow/schemas/config.schema.yaml:27]:
```suggestion
        description: set number of times benchmark should be repeated
```

suggestion for simplification of the descriptions

**Review by Smeds — APPROVED** [review]

File: workflow/rules/padd_target_regions.smk (hunk 8 above)
Comment (ljmesi) [reply workflow/rules/padd_target_regions.smk]: This is fixed now.

File: workflow/rules/padd_target_regions.smk (hunk 9 above)
Comment (ljmesi) [reply workflow/rules/padd_target_regions.smk]: This is fixed now as well in a previous commit.

**Review by jonca79 — APPROVED** [review]

**Review by Smeds — APPROVED** [review]

**Pull request merged**